base/src/main/java/proguard/InputReader.java

```diff
@@ -259,18 +285,26 @@ private void readInput(String          messagePrefix,
                 classPathEntry.getName(),
                 filter != null || classPathEntry.isFiltered() ? " (filtered)" : ""
             );
- 
-            // Create a reader that can unwrap jars, wars, ears, jmods and zips.
-            DataEntryReader reader =
-                new DataEntryReaderFactory(configuration.android)
-                    .createDataEntryReader(classPathEntry,
-                            dataEntryReader);
-
-            // Create the data entry source.
-            DataEntrySource source =
-                new DirectorySource(classPathEntry.getFile());
-
-            // Set he feature name for the class files and resource files
+
+            File classPathFile = classPathEntry.getFile();
+            DataEntryReader reader;
+            DataEntrySource source = maybeGetJrtFallback(classPathFile, classPathEntry.isJmod());
+            if (source != null) {
+                reader = dataEntryReader;
```

Comment:
Not sure if `dataEntryReader` can really be used directly here or if some of the logic of `DataEntryReaderFactory#createDataEntryReader` needs to applied here too.